src/Features/Core/Portable/AddImport/CodeActions/AssemblyReferenceCodeAction.cs

```diff
@@ -31,17 +30,16 @@ private class AssemblyReferenceCodeAction : CodeAction
                 private readonly Lazy<string> _lazyResolvedPath;
 
                 public AssemblyReferenceCodeAction(
-                    AssemblyReference reference,
+                    string title,
+                    ReferenceAssemblyWithTypeResult referenceAssemblyWithType,
                     Document document,
-                    SyntaxNode node,
-                    bool placeSystemNamespaceFirst)
+                    ImmutableArray<TextChange> textChanges)
                 {
-                    _reference = reference;
+                    _referenceAssemblyWithType = referenceAssemblyWithType;
                     _document = document;
-                    _node = node;
-                    _placeSystemNamespaceFirst = placeSystemNamespaceFirst;
+                    _textChanges = textChanges;
 
-                    _title = $"{reference.provider.GetDescription(reference.SearchResult.NameParts)} ({string.Format(FeaturesResources.from_0, reference._referenceAssemblyWithType.AssemblyName)})";
```

Comment:
an example of how we were going back to the provider.  This will in the future just be computed in the OOP process and passed back over.